server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java

```diff
@@ -145,4 +147,16 @@ static ClusterState createDataStream(MetadataCreateIndexService metadataCreateIn
         return ClusterState.builder(currentState).metadata(builder).build();
     }
 
+    public static void validateTemplateForDataStream(String dataStreamName, Metadata metadata) {
+        final String v2Template = MetadataIndexTemplateService.findV2Template(metadata, dataStreamName, false);
+        if (v2Template == null) {
+            throw new IllegalArgumentException("no matching index template found for data stream [" + dataStreamName + "]");
+        }
+        IndexTemplateV2 indexTemplateV2 = metadata.templatesV2().get(v2Template);
+        if (indexTemplateV2.getDataStreamTemplate() == null) {
+            throw new IllegalArgumentException("matching index template [" + v2Template + "] for data stream [" + dataStreamName  +
+                "] has no data stream template");
+        }
```

Comment:
Should we enforce that the timestamp field on the create data stream request is the same as the timestamp field in the data stream template? This can be different if a user provided a different timestamp field via the create data stream api.

If we do enforce this then I think that we should remove the timestamp field from the create data stream request and always use the timestamp field provided in the data stream template inside a itv2.

Reply:
We should at least make specifying the timestamp field non-mandatory. In the future, it should only be necessary to specify if there is no matching template.

Also, the requirement to have the template have a valid mapping for the timestamp field makes it questionable if specifying a timestamp field on the data stream has any real value. Once/if we add the ability to also specify settings and mappings when creating data streams (i.e., create a data stream without any template requirement), it makes more sense to also be able to specify the timestamp field.

I would opt to remove the ability to specify the timestamp field for the first iteration of data streams.

Reply:
Thanks Henning, I will remove the timestamp_field parameter from the create data stream request as part of this PR.